### PR TITLE
PROD-549: Use Django Oauth Toolkit for authentication.

### PR DESCRIPTION
**Pull request by azarembok**
Changes the authentication method used within edx-video-pipeline to use Django Oauth Toolkit (DOT) for authentication, due to the previous method (DOP) being deprecated.

### CODE REVIEW

File: control/veda_val.py [hunk 1]
```diff
@@ -9,7 +9,10 @@
 import urllib3
 import ast
 import json
+import datetime
 
+from django.core.cache import cache
+from edx_rest_api_client.client import EdxRestApiClient
```
Comment (doctoryes) [inline control/veda_val.py]: `EdxRestApiClient` is actually deprecated.
See https://github.com/edx/edx-video-worker/pull/55#discussion_r328193241

Comment (azarembok) [reply control/veda_val.py]: @doctoryes I have updated to use OAuthAPIClient, please review when you get a chance.  Thanks!

File: control/tests/test_val.py [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
 
-from mock import PropertyMock, patch
+from mock import Mock, PropertyMock, patch
 
 import requests
 import urllib3
 import responses
+import datetime
+
+from edx_rest_api_client.client import EdxRestApiClient
```
Comment (doctoryes) [inline control/tests/test_val.py]: This line can be removed.

File: control/veda_val.py [hunk 3]
```diff
@@ -202,14 +181,7 @@ def send_val_data(self):
...
 
-        r1 = requests.get(
-            '/'.join((
-                self.auth_dict['val_api_url'],
-                self.video_proto.val_id
-            )),
-            headers=self.headers,
-            timeout=self.auth_dict['global_timeout']
-        )
+        r1 = self.oauth2_client.request('GET', '/'.join((self.auth_dict['val_api_url'], self.video_proto.val_id, '')))
```
Comment (doctoryes) [inline control/veda_val.py]: The trailing `/` didn't exist before - is this OK?

File: control/veda_val.py [hunk 4]
```diff
@@ -332,13 +304,7 @@ def send_404(self):
...
         )
 
-        r2 = requests.post(
-            self.auth_dict['val_api_url'] + '/',
-            data=json.dumps(sending_data),
-            headers=self.headers,
-            timeout=self.auth_dict['global_timeout']
-        )
-
+        r2 = self.oauth2_client.request('POST', self.auth_dict['val_api_url'] + '/', data=json.dumps(sending_data))
```
Comment (doctoryes) [inline control/veda_val.py]: Nit: It'd be nice to keep a standard way of forming the URL string in all these calls - all using `join`.

File: control/veda_val.py [hunk 5]
```diff
@@ -58,54 +60,33 @@ def __init__(self, video_proto, val_status, **kwargs):
...
-
-        if r.status_code != 200:
-            LOGGER.error('[API] : VAL Token generation')
-            return
-
-        self.val_token = ast.literal_eval(r.text)['access_token']
-        self.headers = {
-            'Authorization': 'Bearer ' + self.val_token,
-            'content-type': 'application/json'
-        }
```
Comment (doctoryes) [inline control/veda_val.py]: 👍 for this code simply melting away...

File: control/veda_val.py (hunk 3 above)
Comment (azarembok) [reply control/veda_val.py]: Thanks for catching this...I don't think it hurts but it doesn't need to be there.  I will remove.

File: control/veda_val.py (hunk 4 above)
Comment (azarembok) [reply control/veda_val.py]: 👍

**Pull request merged**